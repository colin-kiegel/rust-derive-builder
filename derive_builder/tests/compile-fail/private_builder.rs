#[macro_use]
extern crate derive_builder;

pub mod foo {
    /// The builder struct's declaration of privacy should override the field's
    /// attempt to be public later on.
    #[derive(Debug, PartialEq, Default, Builder, Clone)]
    #[builder(private, setter(into))]
    pub struct Lorem {
        pub private: String,
        #[builder(public)]
        pub public: String,
    }
}

fn main() {
    let x = foo::LoremBuilder::default()
    //~^ ERROR struct `LoremBuilder` is private
        .public("Hello")
        .build()
    //~^ ERROR method `build` is private
        .unwrap();

    assert_eq!(x.public, "Hello".to_string());
}
