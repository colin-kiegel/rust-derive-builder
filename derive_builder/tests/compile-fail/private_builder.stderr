error[E0603]: struct `LoremBuilder` is private
  --> $DIR/private_builder.rs:17:18
   |
17 |     let x = foo::LoremBuilder::default()
   |                  ^^^^^^^^^^^^

error[E0624]: method `build` is private
  --> $DIR/private_builder.rs:19:10
   |
19 |         .build()
   |          ^^^^^
